memory_linux.go

```diff
@@ -27,3 +28,18 @@ func sysFreeMemory() uint64 {
 	// So we always convert to uint64 to match signature.
 	return uint64(in.Freeram) * uint64(in.Unit)
 }
+
+func sysAvailableMemory() uint64 {
+	in := &syscall.Sysinfo_t{}
+	err := syscall.Sysinfo(in)
+	if err != nil {
+		return 0
+	}
+	// If this is a 32-bit system, then these fields are
+	// uint32 instead of uint64.
+	// So we always convert to uint64 to match signature.
+	// Buffer/cache ram is included on linux since the kernel
+	// will free this memory for applications if needed, and tends
+	// to use almost all free memory for itself when it can.
+	return (uint64(in.Freeram) + uint64(in.Bufferram)) * uint64(in.Unit)
```

Comment:
On Linux it's generally better to use `MemAvailable` from `/proc/meminfo`. There is a [good justification for this in the Linux kernel git commit](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773).

The hard part is there's no simple stdlib syscall that I am aware of that can read this. :slightly_frowning_face: